APIdefinition.md

```diff
@@ -0,0 +1,96 @@
+# ISIS Project public API
+
+The public API is considered to be all the interfaces that the ISIS
+project provides to the community from which the ISIS user base can
+build more complex tools. This API therefore includes code-level
+structures accessible via software calls to libraries, but also
+more user-facing structures like command-line programs and graphical
+user interface (GUI) programs. Any "interface" via which a user can
+reasonably interact with components of the ISIS project should be
+considered part of the "ISIS Project public API" for the purposes
+of [Semantic Versioning](https://semver.org).
```

Comment:
and here I agree, so perhaps just push the GUI part down into here.

Reply:
Does the concept of a "GUI" need to be added to this sentence?